### PR TITLE
Fix solve

### PR DESCRIPTION
**Pull request by franziska-denk**
I modified PLU factorization such that it works on sparse matrices and also with only 0 and negative entries. Didn't encounter a case unit now where it doesn't work (expect when a column has only 0 entries but then PLU-factorization in generel isn't possible and a ValueError is raised). Solve should now work again in the base.py file of the simplex method. The quadratic tests also did work when I tried them.
The PLU factorization also got a little bit more compact and needs less loops and operations to find the (hopefully) correct solution.

### CODE REVIEW

**Comment by franziska-denk** [comment]
#22

**Comment by tmpr** [comment]
Closes #22

File: shared/solve_linear_system.py [hunk 1]
```diff
@@ -24,25 +24,51 @@ def cholesky_factorization(A):
...
+
+        # adjust permutation matrix
+        if np.isclose(U[i, i], 0.0):
+            max_idxs = A[i:n, i].argsort()[::-1]
+
+            # loop through indices sorted according to descending values, take first index that is non-zero
+            j = 0
+            while A[i:n, i][max_idxs[j]] == 0:
+                if j == (len(max_idxs) - 1):
+                    print(A[i:n, i])
```
Comment (rendstrasser) [inline shared/solve_linear_system.py:53]: Das print ist vermutlich obsolet?

**Comment by rendstrasser** [comment]
Kannst du im simplex.ipynb bei Implemented oben noch einen Punkt dazumachen für die LU-Faktorisierung (considerations from section 13.4)?

**Review by rendstrasser — APPROVED** [review]

**Pull request merged**